src/target/riscv/riscv-013.c

```diff
@@ -4274,62 +4376,58 @@ static int read_memory(struct target *target, target_addr_t address,
 	if (count == 0)
 		return ERROR_OK;
 
-	if (size != 1 && size != 2 && size != 4 && size != 8 && size != 16) {
+	if (!IS_PWR_OF_2(size) || size > 16) {
```

Comment:
The macro `IS_PWR_OF_2` in align.h is buggy because it evaluates to true even for `0` which is not a power of 2.

I have submitted this little fix to upstream (https://review.openocd.org/c/openocd/+/8511) but it will take some time for it to propagate back to riscv-openocd.

For that reason, please, consider extending the check:

```suggestion
	if (!IS_PWR_OF_2(size) || size < 1 || size > 16) {
```

Reply:
Addressed